pallets/parachain-staking/src/types.rs

```diff
@@ -211,100 +209,60 @@ where
 	}
 }
 
-#[derive(Encode, Decode, RuntimeDebug, PartialEq, TypeInfo, MaxEncodedLen)]
-#[scale_info(skip_type_params(MaxCollatorsPerDelegator))]
-#[codec(mel_bound(AccountId: MaxEncodedLen, Balance: MaxEncodedLen))]
-pub struct Delegator<AccountId: Eq + Ord, Balance: Eq + Ord, MaxCollatorsPerDelegator: Get<u32>> {
-	pub delegations: OrderedSet<Stake<AccountId, Balance>, MaxCollatorsPerDelegator>,
-	pub total: Balance,
-}
-
-impl<AccountId, Balance, MaxCollatorsPerDelegator> Delegator<AccountId, Balance, MaxCollatorsPerDelegator>
+pub type Delegator<AccountId, Balance> = Stake<Option<AccountId>, Balance>;
```

Comment:
We need to wrap the collator accountid into an option to differentiate between `default` (empty) delegator states. Unfortunately (or luckily!), `AccountId` does and should not implement `Default` anymore.